File: application-2.1/tek-sorumluluk-python/Kullanici.py
from Adres import Adres

class Kullanici:
    def __init__(self, isim="", mail=""):
        self.isim = isim
        self.mail = mail
        self.adres = None  # Başlangıçta adres boş 

    def set_isim(self, isim):
        self.isim = isim

    def get_isim(self):
        return self.isim

    def set_mail(self, mail):
        self.mail = mail

    def get_mail(self):
        return self.mail

    def adres_guncelle(self, adres):
        if isinstance(adres, Adres):
            self.adres = adres
        else:
            raise TypeError("Adres nesnesi verilmelidir.")

    def get_adres(self):
        return str(self.adres) if self.adres else "Adres bilgisi yok"

    def giris_yap(self, mail, sifre):
        print(f"{self.mail} ile giriş yapıldı.")

    def __str__(self):
        return f"{self.get_isim()} {self.get_mail()} {self.get_adres()}"

File: application-2.1/tek-sorumluluk-c#/tekSorumlulukAdres/Kullanici.cs
public class Kullanici
{
    public string Isim { get; set; }
    public string Mail { get; set; }
    private Adres? adres; //nullable olarak tanımladık

    public Kullanici(string isim = "", string mail = "")
    {
        Isim = isim;
        Mail = mail;
    }

    public void AdresGuncelle(Adres yeniAdres)
    {
        adres = yeniAdres;
    }

    public string GetAdres()
    {
        return adres != null ? adres.ToString() : "Adres bilgisi yok";
    }

    public void GirisYap(string mail, string sifre)
    {
        Console.WriteLine($"{Mail} ile giriş yapıldı.");
    }

    public override string ToString()
    {
        return $"{Isim} {Mail} {GetAdres()}";
    }
}
